experiments/mountain_car_plots.py

```diff
@@ -0,0 +1,269 @@
+    gen_trajs: Optional[List[util.rollout.Trajectory]] = None,
+    exp_trajs: Optional[List[util.rollout.Trajectory]] = None,
+    legend_on: bool = True,
+    title: bool = None,
+    heatmap: bool = True,
+    filter_trans_by_act: bool = True,
+) -> None:
+  """
+  Make a MountainCar heatmap of rewards for a particular action. X axis
+  is position. Y axis is velocity.
+
+  Args:
+    act: The MountainCar action number whose reward we are evaluating.
+      Should be 0, 1, or 2.
+    reward_fn: Reward function. Should accept unnormalized inputs.
+    pos_step_size: The x axis granularity.
+    velocity_step_size: The y axis granularity.
+    mark_goal: If True then plot a line for goal position.
+    gen_trajs: A list of generator trajectories to
+      scatterplot on top of the heatmap.
+    exp_trajs: A list of exp trajectories to scatterplot on
+      top of the heatmap.
+    legend_on: Whether to plot the legend.
+    title: Custom title.
+    heatmap: Whether to plot the heatmap.
+    filter_trans_by_act: If True, then filter out transitions from
+      `gen_trajs` and `exp_trajs` that don't use action `act` before
+      scatterplotting.
+  """
+  assert 0 <= act <= 2
+
+  obs_vec, acts_vec = [], []
+  for p in np.arange(MC_POS_MIN, MC_POS_MAX + 1e-5, step=pos_step_size):
+    for v in np.arange(MC_VEL_MIN, MC_VEL_MAX + 1e-5, step=velocity_step_size):
+      obs_vec.append([p, v])
+      acts_vec.append(act)
+
+  obs_vec = np.array(obs_vec)
+  acts_vec = np.array(acts_vec)
+  next_obs_vec = make_next_obs(obs_vec, acts_vec)
+
+  R = reward_fn(obs_vec, acts_vec, next_obs_vec, None)
+
+  df = pd.DataFrame()
+  df["position"] = np.round(obs_vec[:, 0], 5)
+  df["velocity"] = np.round(obs_vec[:, 1], 5)
+  df["reward"] = R
+  df = df.pivot("velocity", "position", "reward")
+
+  def convert_to_rc(pos, vel):
+    x = (np.array(pos) - MC_POS_MIN) / pos_step_size + 0.5
+    y = (np.array(vel) - MC_VEL_MIN) / velocity_step_size + 0.5
+    return x, y
+
+  def convert_to_rc_filtered(trajs):
+    trans = util.rollout.flatten_trajectories(trajs)
+    obs = trans.obs
+    if filter_trans_by_act:
+      obs = obs[trans.acts == act]
+    pos, vel = obs[:, 0], obs[:, 1]
+    return convert_to_rc(pos, vel)
+
+  plt.figure()
+  if heatmap:
+    sns.heatmap(df)
+
+  if mark_goal:
+    goal_r, _ = convert_to_rc(MC_GOAL_POS, 0)
+    plt.axvline(x=goal_r, linestyle='--',
+                label=f"goal state (pos={MC_GOAL_POS})")
+  if exp_trajs is not None:
+    X, Y = convert_to_rc_filtered(exp_trajs)
+    plt.scatter(X, Y, marker="o", label="expert samples", alpha=0.2)
+  if gen_trajs is not None:
+    X, Y = convert_to_rc_filtered(gen_trajs)
+    plt.scatter(X, Y, marker="o", c="yellow", label="policy samples",
+                alpha=0.2)
+
+  act_names = ["left", "neutral", "right"]
+  if title is None:
+    title = f"Action {act_names[act]}"
+  plt.title(title)
+  if legend_on:
+    plt.legend(loc="center left", bbox_to_anchor=(0, 1.3))
+
```

Comment:
```suggestion
    return fig
```

Clearer if more functional style. (You can do things like create multiple heatmaps, put the figures in a dict, save them later...)